src/EnergyPlus/Plant/Solution/PlantLoopSolver.cc

```diff
@@ -168,215 +158,248 @@ namespace PlantLoopSolver {
+        }
 
 		// A couple things are specific to which LoopSide we are on
-		if ( LoopSideNum == DemandSide ) {
+		if ( LoopSideNum == DataPlant::DemandSide ) {
 
 			// Pass the loop information via the HVAC interface manager
-			UpdatePlantLoopInterface( LoopNum, LoopSideNum, PlantLoop( LoopNum ).LoopSide( DemandSide ).NodeNumOut, PlantLoop( LoopNum ).LoopSide( SupplySide ).NodeNumIn, ReSimOtherSideNeeded, PlantLoop( LoopNum ).CommonPipeType );
+			HVACInterfaceManager::UpdatePlantLoopInterface( LoopNum, LoopSideNum, thisPlantLoop.LoopSide( DataPlant::DemandSide ).NodeNumOut, thisPlantLoop.LoopSide( DataPlant::SupplySide ).NodeNumIn, ReSimOtherSideNeeded, thisPlantLoop.CommonPipeType );
 
 		} else { //LoopSide == SupplySide
 
 			// Update pressure drop reporting, calculate total loop pressure drop for use elsewhere
-			SimPressureDropSystem( LoopNum, FirstHVACIteration, PressureCall_Update );
+			PlantPressureSystem::SimPressureDropSystem( LoopNum, FirstHVACIteration, DataPlant::PressureCall_Update );
 
 			// Pass the loop information via the HVAC interface manager (only the flow)
-			UpdatePlantLoopInterface( LoopNum, LoopSideNum, PlantLoop( LoopNum ).LoopSide( SupplySide ).NodeNumOut, PlantLoop( LoopNum ).LoopSide( DemandSide ).NodeNumIn, ReSimOtherSideNeeded, PlantLoop( LoopNum ).CommonPipeType );
+			HVACInterfaceManager::UpdatePlantLoopInterface( LoopNum, LoopSideNum, thisPlantLoop.LoopSide( DataPlant::SupplySide ).NodeNumOut, thisPlantLoop.LoopSide( DataPlant::DemandSide ).NodeNumIn, ReSimOtherSideNeeded, thisPlantLoop.CommonPipeType );
 
 			// Update the loop outlet node conditions
-			CheckLoopExitNode( LoopNum, FirstHVACIteration );
+			thisPlantLoop.loopSolver.CheckLoopExitNode( LoopNum, FirstHVACIteration );
 
 		}
 
 		// Update some reporting information at Plant half loop level
-		UpdateLoopSideReportVars( LoopNum, LoopSideNum, InitialDemandToLoopSetPointSAVED, LoadToLoopSetPointThatWasntMet );
+		thisPlantLoop.loopSolver.UpdateLoopSideReportVars( LoopNum, LoopSideNum, InitialDemandToLoopSetPointSAVED, LoadToLoopSetPointThatWasntMet );
 
 	}
 
-	//==================================================================!
-	//==================================================================!
-	//==================================================================!
-
-	//==================================================================!
-	//================= TOPOLOGY VALIDATION ROUTINE ====================!
-	//==================================================================!
+	void
+	PlantLoopSolverClass::TurnOnAllLoopSideBranches(
+			DataPlant::HalfLoopData & loop_side
+	)
+	{
+		for ( int branchNum = 2; branchNum <= loop_side.TotalBranches-1; ++branchNum ) {
+			auto & branch = loop_side.Branch( branchNum );
+			branch.QuickDisableForCSBranchPumping = false;
+		}
+	}
 
-	m_FlowControlValidator
-	ValidateFlowControlPaths(
+    void
```

Comment:
This function is to be called after a full load/flow solution has been achieved on the loopside, but there are constant speed branch pumps present that could be causing branches to turn on when they aren't needed.  It loops over the branches and components, checks load, and if there isn't any load on the loop, it disables the branch to allow a second pass.